src/chessboard/components/Square.tsx

```diff
@@ -69,6 +69,8 @@ export function Square({
       lastPieceColour,
     ]
   );
+  const [lastSquareDraggedOver, setLastSquareDraggedOver] =
+    useState<Sq | null>(null);
```

Comment:
Hmmm `Square` component represents one square. currently you have been added 64 independent states for each Square. Can you explain please why?

Reply:
this may be better placed in the chessboard context

Reply:
The simple solution is not do the check, but then you have the onDragOverSquare() running a million times, and I know in my own project that would kill it - a fair bit happens in that function.
I'll take a look at getting it moved into the Chessboard Context - I think you're right, probably works better there.